syntax = "proto3";
package calculator;

// Math service definition
service MathService {
  rpc Add (AddRequest) returns (AddResponse) {}
  rpc Subtract (SubtractRequest) returns (SubtractResponse) {}
}

// The request message containing two numbers.
message AddRequest {
  int32 num1 = 1;
  int32 num2 = 2;
}

// The response message containing the result.
message AddResponse {
  int32 result = 1;
}

// The request message containing two numbers.
message SubtractRequest {
  int32 num1 = 1;
  int32 num2 = 2;
}

// The response message containing the result.
message SubtractResponse {
  int32 result = 1;
}